src/pubsub/mod_pubsub.erl

```diff
@@ -830,37 +870,39 @@ out_subscription(Acc, _, _, _, _) ->
                       _:: any()) ->
     mongoose_acc:t().
 in_subscription(Acc, User, Server, Owner, unsubscribed, _) ->
-    unsubscribe_user(jid:make(User, Server, <<>>), Owner),
+    unsubscribe_user(mod_pubsub_db_backend, jid:make(User, Server, <<>>), Owner),
     Acc;
 in_subscription(Acc, _, _, _, _, _) ->
     Acc.
 
-unsubscribe_user(Entity, Owner) ->
+unsubscribe_user(Backend, Entity, Owner) ->
     ServerHosts = lists:usort(lists:foldl(
                                 fun(UserHost, Acc) ->
                                         case gen_mod:is_loaded(UserHost, mod_pubsub) of
                                             true -> [UserHost|Acc];
                                             false -> Acc
                                         end
                                 end, [], [Entity#jid.lserver, Owner#jid.lserver])),
-    spawn(fun() -> [unsubscribe_user(ServerHost, Entity, Owner) || ServerHost <- ServerHosts] end).
+    spawn(fun() -> [unsubscribe_user(Backend, ServerHost, Entity, Owner) || ServerHost <- ServerHosts] end).
 
-unsubscribe_user(Host, Entity, Owner) ->
+unsubscribe_user(Backend, Host, Entity, Owner) ->
     BJID = jid:to_lower(jid:to_bare(Owner)),
     lists:foreach(fun (PType) ->
-                         unsubscribe_user_per_plugin(Host, Entity, BJID, PType)
+                         unsubscribe_user_per_plugin(Backend, Host, Entity, BJID, PType)
                   end, plugins(Host)).
 
-unsubscribe_user_per_plugin(Host, Entity, BJID, PType) ->
-    {result, Subs} = node_action(Host, PType, get_entity_subscriptions, [Host, Entity]),
+%% TODO: duplicated logic maybe?
```

Comment:
could you elaborate more on that?

Reply:
that is code for the exported function `unsubscribe_user`, but we wrote a very similar functionality for our `remove_user`, where the only difference is basically whether backend is loaded or we're trying with all of them. I thought we could review that.

Reply:
yeah, if you can reuse existing functions, lets do that

Reply:
But this requires a bit more rework I didn't have the time to do in this effort, can you help me out with that one?